web/src/main/java/org/springframework/security/web/firewall/RequestRejectedHandler.java

```diff
@@ -42,4 +47,12 @@
 	void handle(HttpServletRequest request, HttpServletResponse response,
 			RequestRejectedException requestRejectedException) throws IOException, ServletException;
 
+	/**
+	 * Determines if this handler should be invoked.
+	 * First available handler is invoked if there are multiple suitables ones.
+	 * @param request that resulted in an <code>RequestRejectedException</code>
+	 */
+	default boolean shouldHandle(HttpServletRequest request) {
```

Comment:
Instead of a `shouldHandle` method, it seems like the `handle` method can decide whether or not to participate and then simply return early.

Reply:
Reverted